packages/firebase_remote_config/lib/src/remote_config.dart

```diff
@@ -0,0 +1,210 @@
+      final RemoteConfigValue remoteConfigValue =
+          new RemoteConfigValue._(value['value'].cast<int>(), valueSource);
+      parsedParameters[key] = remoteConfigValue;
+    });
+    return parsedParameters;
+  }
+
+  /// Set the configuration settings for this RemoteConfig instance.
+  ///
+  /// This can be used for enabling developer mode.
+  Future<void> setConfigSettings(
+      RemoteConfigSettings remoteConfigSettings) async {
+    await _channel
+        .invokeMethod('RemoteConfig#setConfigSettings', <String, dynamic>{
+      'debugMode': remoteConfigSettings.debugMode,
+    });
+    _remoteConfigSettings = remoteConfigSettings;
+    return new Future<void>.value();
+  }
+
+  /// Fetches parameter values for your app. Parameter values may be from
+  /// Default Config (local cache) or Remote Config if enough time has elapsed
+  /// since parameter values were last fetched from the server. The default
+  /// expiration time is 12 hours. Expiration must be defined in seconds.
+  Future<void> fetch({Duration expiration: const Duration(hours: 12)}) async {
+    try {
+      final Map<dynamic, dynamic> properties = await _channel.invokeMethod(
+          'RemoteConfig#fetch',
+          <dynamic, dynamic>{'expiration': expiration.inSeconds});
+      _lastFetchTime =
+          new DateTime.fromMillisecondsSinceEpoch(properties[lastFetchTimeKey]);
+      _lastFetchStatus = LastFetchStatus.values[properties[lastFetchStatusKey]];
+    } on PlatformException catch (e) {
+      _lastFetchTime =
+          new DateTime.fromMillisecondsSinceEpoch(e.details[lastFetchTimeKey]);
+      _lastFetchStatus = LastFetchStatus.values[e.details[lastFetchStatusKey]];
+      if (e.code == RemoteConfig.fetchFailedThrottledKey) {
+        print('fetch failed throttled');
+        final int fetchThrottleEnd = e.details['FETCH_THROTTLED_END'];
+        throw new FetchThrottledException._(endTimeInMills: fetchThrottleEnd);
+      } else {
+        print('fetch failed unknown');
+        throw new Exception('Unable to fetch remote config');
+      }
+    }
+    print('fetch succeeded');
+    return new Future<void>.value();
+  }
+
+  /// Activates the fetched config. This makes fetched key-values take effect.
+  ///
+  /// The returned Future contains true if the fetched config is different
+  /// from the currently activated config, it contains false otherwise.
+  Future<bool> activateFetched() async {
+    final Map<dynamic, dynamic> rawParameters =
+        await _channel.invokeMethod('RemoteConfig#activate');
+    final Map<String, RemoteConfigValue> fetchedParameters =
+        _parseRemoteConfigParameters(parameters: rawParameters);
+    final MapEquality<String, RemoteConfigValue> mapEquality =
+        const MapEquality<String, RemoteConfigValue>(
+      keys: const Equality<String>(),
+      values: const Equality<RemoteConfigValue>(),
+    );
+    final bool newConfig = mapEquality.equals(_parameters, fetchedParameters);
```

Comment:
I'm not sure that this will work, since you haven't implemented an `operator==` for `RemoteConfigValue`. Rather than compute this boolean ourselves, can we check the boolean provided by the native call to `activateFetched` and just pass a `null` parameters map to Dart if nothing changed?

Reply:
Totally agree we should get this value from the native side.

Reply:
Done